src/main/java/elrond/ProxyProvider.java

```diff
@@ -45,13 +47,49 @@ public AccountOnNetwork getAccount(Address address) throws IOException, AddressE
         return AccountOnNetwork.fromProviderPayload(payload);
     }
 
+    public ESDTDataResponse getESDTData(Address address, String tokenIdentifier) throws IOException, AddressException {
+        String hexTokenIdentifier = this.adjustTokenIdentifier(tokenIdentifier);
+        String responseJson = this.doGet(String.format("accounts/%s/tokens/%s", address.bech32(), hexTokenIdentifier));
+        ESDTDataResponse typedResponse = gson.fromJson(responseJson, ESDTDataResponse.class);
+        if (typedResponse == null || typedResponse.balance == null) {
+            return ESDTDataResponse.empty();
+        }
+
+        return typedResponse;
+    }
+
+    public BigInteger getNFTBalance(Address address, String tokenIdentifier, long nonce) throws IOException, AddressException {
+        String hexTokenIdentifier = this.adjustTokenIdentifier(tokenIdentifier);
+        String nonceStr = String.format("%d", nonce);
+        if(nonceStr.length() % 2 == 1 ) {
```

Comment:
this check was done with hex, is it needed also for integer strings or should the request be made with hex value?

Reply:
Good catch. Fixed

Reply:
ok